presto-raptor/src/main/java/com/facebook/presto/raptor/TemporalFunction.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.raptor;
+
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.type.DateType;
+import com.facebook.presto.spi.type.TimestampType;
+import com.facebook.presto.spi.type.Type;
+
+import static com.facebook.presto.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
+import static java.lang.String.format;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+public interface TemporalFunction
+{
+    int getDay(Block temporalBlock, int position);
+
+    static TemporalFunction create(Type temporalColumnType)
+    {
+        if (temporalColumnType == DateType.DATE) {
+            return (temporalBlock, position) -> temporalBlock.getInt(position, 0);
+        }
+
+        if (temporalColumnType == TimestampType.TIMESTAMP) {
+            return (temporalBlock, position) -> (int) MILLISECONDS.toDays(temporalBlock.getLong(position, 0));
+        }
+
+        throw new PrestoException(GENERIC_INTERNAL_ERROR, format("Wrong type %s for temporalColumnType", temporalColumnType));
```

Comment:
It's best to put the value at the end of the error message if there's only value (easier to grep the logs, etc.)